tests/test_build.py

```diff
@@ -367,28 +367,24 @@ def mock_util_call(cmd):
     def test_parse_build_results_failed_pats(self):
         """
         Test parse_build_results with a test log indicating failure due to a
-        missing testpkg (failed pat error). The specific error is a python
-        ImportError.
+        missing package.
         """
-        def mock_util_call(cmd):
-            del cmd
-
         build.config.setup_patterns()
         call_backup = build.util.call
-        build.util.call = mock_util_call
+        build.util.call = MagicMock(return_value=None)
         fm = files.FileManager()
 
-        open_name = 'build.util.open_auto'
-        content = 'line 1\nImportError: No module named testpkg\nexiting'
-        m_open = mock_open(read_data=content)
-
-        with patch(open_name, m_open, create=True):
-            build.parse_build_results('testname', 0, fm)
-
-        build.util.call = call_backup
-
-        self.assertIn('testpkg-python', build.buildreq.buildreqs)
-        self.assertEqual(build.must_restart, 1)
+        with open('tests/builderrors', 'r') as f:
+            builderrors = f.readlines()
+            for error in builderrors:
+                if not error.startswith('#'):
+                    input, output = error.strip('\n').split('|')
+                    build.buildreq.buildreqs = set()
+                    build.util.open_auto = mock_open(read_data=input)
+                    build.parse_build_results('testname', 0, fm)
+
+                    self.assertIn(output, build.buildreq.buildreqs)
+                    self.assertGreater(build.must_restart, 0)
```

Comment:
I am a bit worried that build.must_restart is 0 since a buildreq was added. Am I missing something?

Reply:
It's an `assertGreater`, so it shouldn't be 0 at this point, right?

Reply:
Indeed!